repose-aggregator/components/filters/header-translation-filter/src/main/scala/org/openrepose/filters/headertranslation/HeaderTranslationFilter.scala

```diff
@@ -74,10 +74,19 @@ class HeaderTranslationFilter @Inject()(configurationService: ConfigurationServi
         if (originalHeaderValues.nonEmpty) {
           sourceHeader.getNewName foreach { newHeaderName =>
             originalHeaderValues foreach { originalHeaderValue =>
-              if (quality.isEmpty) {
-                httpRequest.addHeader(newHeaderName, originalHeaderValue)
+              def writeHeaders(writeWithoutQuality: (String, String) => Unit,
+                               writeWithQuality: (String, String, Double) => Unit): Unit = {
+                if (quality.isEmpty) {
+                  writeWithoutQuality(newHeaderName, originalHeaderValue)
+                } else {
+                  writeWithQuality(newHeaderName, withoutQuality(originalHeaderValue), quality.get)
+                }
+              }
```

Comment:
It totally does, but in doing so introduces another layer of abstraction (and another call on the stack). It seems like trading one minor form of complexity for another, which is why in the comment below I said that it probably comes down to preference.

Anywhoo, if you're happy with it, I have no reason not to be.